#! /bin/sh /usr/share/dpatch/dpatch-run
## msa090002.dpatch by Francois Marier <francois@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: protect user profile images if $CFG->forcelogin enabled

@DPATCH@
diff --git a/user/pix.php b/user/pix.php
index b3509ef..d37d18b 100644
--- a/user/pix.php
+++ b/user/pix.php
@@ -3,11 +3,15 @@
       // Syntax:   pix.php/userid/f1.jpg or pix.php/userid/f2.jpg
       //     OR:   ?file=userid/f1.jpg or ?file=userid/f2.jpg
 
-    $nomoodlecookie = true;     // Because it interferes with caching
-
     require_once('../config.php');
     require_once($CFG->libdir.'/filelib.php');
 
+    if (!empty($CFG->forcelogin) and !isloggedin()) {
+        // protect images if login required and not logged in;
+        // do not use require_login() because it is expensive and not suitable here anyway
+        redirect($CFG->pixpath.'/u/f1.png');
+    }
+
     // disable moodle specific debug messages
     disable_debugging();
 
